src/compiler/checker.ts

```diff
@@ -11378,6 +11378,18 @@ namespace ts {
         }
 
         function getTypeWithFacts(type: Type, include: TypeFacts) {
+            if (type.flags & TypeFlags.IndexedAccess) {
+                // TODO (weswig): This is a substitute for a lazy negated type to remove the types indicated by the TypeFacts from the (potential) union the IndexedAccess refers to
+                //  - instead of defering the resolution of the access, we apply the facts to the base constraint of the access instead; so this works under most circumstances,
+                //    like when the index refers to an optional member and you want to access that member, but unfortunately if that member's type is supposed to vary with the base,
+                //    then the eager evaluation of this removes that relationship

```

Comment:
1. Can you add a short comment to the test that links the test case to this limitation?
2. Can you put in an example here?
3. Can you use fewer words if you have an example?